plugins/by-name/noice/default.nix

```diff
@@ -1,35 +1,103 @@
 {
   lib,
-  helpers,
-  config,
-  pkgs,
   ...
 }:
-# TODO: This uses a lot of types.anything because noice.nvim types are quite complex.
-# It should be possible to map them to nix, but they would not map really well through
-# toLuaObject, we would maybe need some ad-hoc pre-processing functions.
-with lib;
-{
-  options.plugins.noice = helpers.neovim-plugin.extraOptionsOptions // {
-    enable = mkEnableOption ''
-      noice.nvim, an experimental nvim UI.
-      Note that if treesitter is enabled you need the following parsers:
-        vim, regex, lua, bash, markdown, markdown_inline
-    '';
+let
+  inherit (lib) types;
+  inherit (lib.nixvim) defaultNullOpts;
+in
+lib.nixvim.neovim-plugin.mkNeovimPlugin {
+  name = "noice";
+  originalName = "noice.nvim";
+  package = "noice-nvim";
+
+  maintainers = [ lib.maintainers.khaneliman ];
+
+  description = ''
+    `noice.nvim`, an experimental Neovim UI.
+
+    > [!NOTE] If treesitter is enabled you need the following parsers:
+    >  vim, regex, lua, bash, markdown, markdown_inline
```

Comment:
```suggestion
    > [!NOTE]
    > If treesitter is enabled you need the following parsers:
    > vim, regex, lua, bash, markdown, markdown_inline
```

I'm unsure if your initial syntax is valid, but elsewhere we have the admonition syntax on a separate line from the actual text body

Reply:
Doh, you're right 
![image](https://github.com/user-attachments/assets/aebe093d-08e7-4390-bb7b-ab26231454c4)